src/ilasm/CMakeLists.txt

```diff
@@ -66,14 +77,47 @@ set(ILASM_LINK_LIBRARIES
   utilcodestaticnohost
   mscorpe
   mdhotdata_full
+  ${START_LIBRARY_GROUP} # Start group of libraries that have circular references
+  mdcompiler_wks
+  mdruntime_wks
+  mdruntimerw_wks
+  strongname_tool
+  mdstaticapi
+  ${END_LIBRARY_GROUP} # End group of libraries that have circular references
+  ceefgen
   corguids
 )
 
+if(WIN32)
+    list(APPEND ILASM_LINK_LIBRARIES
+        ${STATIC_MT_CRT_LIB}
+        ${STATIC_MT_VCRT_LIB}
+        mdwinmd_wks
+        kernel32.lib
+        advapi32.lib
+        ole32.lib
+        oleaut32.lib
+        uuid.lib
+        user32.lib
+        version.lib
+        shlwapi.lib
+        bcrypt.lib
+        RuntimeObject.lib
+    )
+else()
+    list(APPEND ILASM_LINK_LIBRARIES
+        ${START_WHOLE_ARCHIVE} # force all PAL objects to be included so all exports are available
```

Comment:
This ${START_WHOLE_ARCHIVE} should not be needed. Only the coreclr needs this as it exports symbols, some of which are from PAL.